lib/ansible/module_utils/basic.py

```diff
@@ -1630,32 +1614,7 @@ def _check_locale(self):
                            to_native(e), exception=traceback.format_exc())
 
     def _handle_aliases(self, spec=None, param=None):
-        # this uses exceptions as it happens before we can safely call fail_json
-        aliases_results = {}  # alias:canon
-        if param is None:
-            param = self.params
-
-        if spec is None:
-            spec = self.argument_spec
-        for (k, v) in spec.items():
-            self._legal_inputs.append(k)
-            aliases = v.get('aliases', None)
-            default = v.get('default', None)
-            required = v.get('required', False)
-            if default is not None and required:
-                # not alias specific but this is a good place to check this
-                raise Exception("internal error: required and default are mutually exclusive for %s" % k)
-            if aliases is None:
-                continue
-            if not isinstance(aliases, SEQUENCETYPE) or isinstance(aliases, (binary_type, text_type)):
-                raise Exception('internal error: aliases must be a list or tuple')
-            for alias in aliases:
-                self._legal_inputs.append(alias)
-                aliases_results[alias] = k
-                if alias in param:
-                    param[k] = param[alias]
-
-        return aliases_results
+        return handle_aliases(self.params, self.argument_spec, self._legal_inputs, spec, param)
```

Comment:
This is a private function (starts with an underscore) so I'd hope we could move this without having to leave a stub for backwards compatibility.  If you want to remove the stub in a separate commit/PR just in case, though, I can live with that.

Reply:
After taking a look at the new function, I could see leaving the stub function in so that it can establish default values for arg_spec and params.